optimum/intel/openvino/quantization.py

```diff
@@ -849,15 +853,22 @@ def _prepare_text_encoder_model_calibration_data(
                 with numpy_seed(self.seed):
                     random_positions = np.random.randint(0, seq_len, num_samples)
 
+            tokenizer = None
             pbar = tqdm(total=num_samples, desc="Collecting calibration data")
             for item in dataset:
-                inputs = tokenizer(item["text"], truncation=True, max_length=seq_len, return_tensors="np")
-                if inputs["input_ids"].shape[1] < seq_len:
-                    continue
+                if "input_ids" in item:
```

Comment:
I've noticed that similar checks will have to be added to other model type dataset preparation functions which is quite inconvenient. This is shortcoming after #1232 . I hope there is a better solution, at least temporary. I need to analyze it a bit more.

Reply:
I have some ideas for improvement, but prototyping is needed. I will have to address this after the upcoming release.